buffer/src/test/java/io/netty/buffer/SlicedByteBufTest.java

```diff
@@ -31,14 +30,18 @@
 public class SlicedByteBufTest extends AbstractByteBufTest {
 
     @Override
-    protected ByteBuf newBuffer(int length, int maxCapacity) {
+    protected final ByteBuf newBuffer(int length, int maxCapacity) {
         Assume.assumeTrue(maxCapacity == Integer.MAX_VALUE);
-        ByteBuf buffer = Unpooled.wrappedBuffer(
-                new byte[length * 2], length > 1 ?
-                        PlatformDependent.threadLocalRandom().nextInt(length - 1) + 1 : 0, length);
-        assertEquals(0, buffer.readerIndex());
-        assertEquals(length, buffer.writerIndex());
-        return buffer;
+        int offset = length == 0 ? 1 : PlatformDependent.threadLocalRandom().nextInt(length);
```

Comment:
if length is 0 we will allocate a buffer of size 0, but take a slice starting from index 1 ... won't this be IOOB?